tmt/steps/prepare/install.py

```diff
@@ -266,15 +266,15 @@ def install_local(self):
 
     def install_from_repository(self):
         """ Install packages from the repository """
-        packages = self.list_packages(self.repository_packages, title="package")
```

Comment:
Could you keep `self.list_packages(self.repository_packages, title="package")` call to print what is being installed? (and ignore its return value. IMO better to ignore the value than to reimplement how it is printed).

Reply:
This is more of an rpm-ostree behavior problem.
With rpm-ostree, there is no --skip-broken flag. So if a list of packages is called, and even one package is already satisfied the whole cmd returns in error and no action is taken.  So using list_packages and the string created will not work because we have to address each package on an individual basis.

unless of course you have a better idea :)

Actually I've worked an improvement already to the rpm-ostree package selection, with a new function. I was going to use a new PR because it has nothing to do with the bug here. But in summary, I have the function sorting packages creating two lists, recommend and required, of packages not already satisfied.

Reply:
I've got an idea, I'll just push the improvement now :)

Reply:
The logic for sort_packages is :
create two lists, because recommended can be left to not satisfy the requirement. Required are ... required. By this logic, if we check that the package is not already installed, then all required packages can be called as one list, because if any should fail, then the requirements are not met regardless. Recommended packages are still handled on a 1 x 1 basis because any of these can be ignored if not successful. Should help improve some time (at least on long required lists)

and I wasn't grasping that the list_packages function prints the action being performed haha, got it now

Reply:
The new method `sort_packages` looks good to me. Thanks for improving this.